services/web/server/src/simcore_service_webserver/projects/projects_ports_handlers.py

```diff
@@ -82,22 +83,46 @@ async def _get_validated_workbench_model(
     return workbench
 
 
-#
+routes = web.RouteTableDef()
+
+
+class _InputSchema:
+    class Config:
+        allow_population_by_field_name = False
+        extra = Extra.forbid
+        allow_mutations = False
+
+
+class _OutputSchema:
+    class Config:
+        allow_population_by_field_name = True
+        extra = Extra.ignore
```

Comment:
Question: would it not make more sense to have `Extra.forbid` here? In the case the outputs schema would change you would get an error, instead of having some unattended side effect because the fields are suppressed but no error in the call chain is raised.

Reply:
The pydantic models used as API outputs are filled using some internal structures. I use this approach to "prune" some extra fields that might be in an internal data model. For instance, assume an output model ``UserGet`` that I init using some RowProxy of the users  database table

```python

# user RowProxy has username, email, password

class UserGet(_OutputSchema):
      username: str
      email: EmailStr

      class Config:
            orm= True


return UserGet.parse_obj(user)
```